mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinListener.java

```diff
@@ -58,25 +64,30 @@ public ExecutionMode getExecutionMode() {
 
     @Override
     public void event(Event event) {
-        LOGGER.debug("Event {} received in listener.", event);
-        if (event instanceof EventFactory.AddedImpl) {
-            EventFactory.AddedImpl addedToMailboxEvent = (EventFactory.AddedImpl) event;
-            if (isEventOnSpamMailbox(addedToMailboxEvent)) {
+        if (event instanceof MessageMoveEvent) {
+            MessageMoveEvent messageMoveEvent = (MessageMoveEvent) event;
+            if (isMessageMovedToSpamMailbox(messageMoveEvent)) {
                 LOGGER.debug("Spam event detected");
-                ImmutableList<InputStream> messages = addedToMailboxEvent.getAvailableMessages()
+                ImmutableList<InputStream> messages = messageMoveEvent.getMessages()
                     .values()
                     .stream()
                     .map(Throwing.function(Message::getFullContent))
                     .collect(Guavate.toImmutableList());
-                spamAssassin.learnSpam(messages, addedToMailboxEvent.getMailboxPath().getUser());
+                spamAssassin.learnSpam(messages, messageMoveEvent.getSession().getUser().getUserName());
```

Comment:
It looks like a bad idea to put the sesion into the Event : we don't want to share it on a MQ and it's probably mutable.

Reply:
I agree with you, lot of things has to be enhanced in Events; but I really think it's not the time todo such modifications.

Reply:
You may have seen I do little comments on PR because we should try to go fast. But in this case, you are sharing a mutable structure in a multithread context, it deserves to find a solution.

Reply:
Then, I will have to modify _MailboxEvent_ and all subclasses... 
should I take the time todo that, IYO?

Reply:
I may be wrong, but Event interface is a new one.
You just need to change the new MessageMoveEvent.

Reply:
BTW, you're true that we already have this bug in former Event type.

Reply:
The only thing is that we need that to retrieve the user spam mailbox id, may be we can give it to the constructor.

Reply:
we also need trash

Reply:
ok, we will do that later, you are right